Sources/Private/Experimental/ValueProviderStore.swift

```diff
@@ -99,3 +107,22 @@ extension AnyValueProviderStorage {
     }
   }
 }
+
+extension AnimationKeypath {
+  /// Whether or not this keypath matches the given keypath from the animation hierarchy
+  func matches(_ keypath: AnimationKeypath) -> Bool {
+    var regex = "^" // match the start of the string
+      + keys.joined(separator: "\\.") // match this keypath, escaping "." characters
+      + "$" // match the end of the string
+
+    // ** wildcards match anything
+    //  - "**.Color" matches both "Layer 1.Color" and "Layer 1.Layer 2.Color"
+    regex = regex.replacingOccurrences(of: "**", with: ".+")
+
+    // * wildcards match any individual path component
+    //  - "*.Color" matches "Layer 1.Color" but not "Layer 1.Layer 2.Color"
+    regex = regex.replacingOccurrences(of: "*", with: "[^.]+")
+
+    return keypath.fullPath.range(of: regex, options: .regularExpression) != nil
+  }
+}
```

Comment:
Do you think we could write some small unit tests on this logic? This method is super well documented but it feels like it might be helpful to test this too as it doesn't seem like the snapshot tests validate single * wildcards

Reply:
That seems like a good idea 👍🏻